packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityPlugin.java

```diff
@@ -76,6 +77,60 @@ private static String getNetworkType(int type) {
         return "none";
     }
   }
+  
+  private static String getNetworkSubType(int type, int subType) {
+    switch (type) {
```

Comment:
I think if we are going to get the network subtype, we should separate the return value for each subtype. For example, for ConnectivityManager.TYPE_ETHERNET we should just return "TYPE_ETHERNET", and we also should match both Android and iOS to give a main type (e.g. TYPE_MOBILE_HIPRI) and a sub type (e.g. NETWORK_TYPE_1xRTT)
Since we are adding a new method now, you are free to not limit the return type to just a String, it can be an object type for example:
```dart
class ConnectionType {
   final String type;
   final String subType;
}

```

I hope this makes sense.